Use Typer instead of argparse

paketb0te opened the pull request:
The PR for #43 :)

initialcommit-io commented:
@paketb0te Thanks for this!  Just 2 comments for now:

- Sorry to ask this since we discussed changing the file names, but I think that might interfere with Git blame by replacing the whole file and its content.  Can you leave the file names as they were (it's fine to keep the Class names updated to the new values), and I'll rename them in a future commit?

- It looks like `self.maxrefs` is still there in some places.  Do you want to remove that?

paketb0te commented:
@initialcommit-io of course, I just realized that I still have both files for each command in my branch (kept the original ones to look at while modifying the new ones).

I moved everything back into the original files now (so that `git blame` works correctly).

Since we introduced a new dependency on typer, we should probably start including something like a `requirements.txt` (or does setup.py automatically install dependencies from PyPi? :thinking:)

initialcommit-io commented:
@paketb0te Thanks! Setup.py does automatically include the dependencies, so we should be ok for now, but we do still have the other open issue for the dependency system upgrade...
